includes/Abilities/Title_Generation/Title_Generation.php

```diff
@@ -204,4 +224,43 @@ protected function meta(): array {
 			'show_in_rest' => true,
 		);
 	}
+
+	/**
+	 * Generates title suggestions from the given content.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param string|array<string, string> $context The context to generate a title from.
+	 * @param int $candidates The number of titles to generate.
+	 * @return array<string>|\WP_Error The generated titles, or a WP_Error if there was an error.
+	 */
+	protected function generate_titles( $context, int $candidates = 1 ) {
+		// Convert the context to a string if it's an array.
+		if ( is_array( $context ) ) {
+			$context = implode(
+				"\n",
+				array_map(
+					static function ( $key, $value ) {
+						return sprintf(
+							'%s: %s',
+							ucwords( str_replace( '_', ' ', $key ) ),
+							$value
+						);
+					},
+					array_keys( $context ),
+					$context
+				)
+			);
+		}
+
+		// Get the prompt builder.
+		$prompt_builder = AI_Client::prompt_with_wp_error( '"""' . $context . '"""' )
+			->using_system_instruction( $this->get_system_instruction() )
+			->using_temperature( 0.7 )
+			->using_candidate_count( (int) $candidates )
+			->using_model_preference( ...get_preferred_models() );
+
+		// Make the request.
+		return $prompt_builder->generate_texts();
```

Comment:
Comes together nicely! 😃 
```suggestion
		// Generate the text candidates using the AI
		return AI_Client::prompt_with_wp_error( '"""' . $context . '"""' )
			->using_system_instruction( $this->get_system_instruction() )
			->using_temperature( 0.7 )
			->using_candidate_count( (int) $candidates )
			->using_model_preference( ...get_preferred_models() );
			>generate_texts();
```

Reply:
Updated in https://github.com/WordPress/ai/pull/67/commits/919b6609ff62ee19237819cbc9473a2c7690d2e6